atlascope/core/rest/endpoints/tile_endpoints.py

```diff
@@ -19,7 +20,11 @@ class TileMetadataView(GenericAPIView, mixins.RetrieveModelMixin):
 
     def get(self, *args, **kwargs):
         dataset = self.get_object()
-        tile_source = GDALFileTileSource(f'/vsicurl/use_head=no&url={dataset.content.url}')
+        cached = fsspec.open_local(
+            f'simplecache::{dataset.content.url}',
+            filecache={'cache_storage': '/tmp/files'},
+        )
+        tile_source = OMETiffFileTileSource(cached[0])
```

Comment:
We'll likely receive 10+ separate requests instantaneously for a dataset. Will all 10+ of those requests download a file? Or is there a lock so that only 1 does?

Reply:
Each request would download a file with one channel. That's the crux of the tradeoff between compositing the channels client side versus server side. Which do you think is better for our use case?

Reply:
Oh I see. Yes, your description is in line with what I read from the docs, but I'm not sure how it handles races between parallel requests. It's something that we can come back to, like you say. 

Let's explore what it would look like to do client-side compositing first and fall back to server-side compositing, since I've tried that and I know how to implement it. I think it would be nice to allow the user to toggle channels and change colors without making subsequent requests.